file/file_util.cc

```diff
@@ -72,6 +65,30 @@ IOStatus CopyFile(FileSystem* fs, const std::string& source,
   return dest_writer->Sync(use_fsync);
 }
 
+IOStatus CopyFile(FileSystem* fs, const std::string& source,
+                  const std::string& destination, uint64_t size, bool use_fsync,
+                  const std::shared_ptr<IOTracer>& io_tracer,
+                  const Temperature temperature) {
```

Comment:
The comment in FileOptions suggests that temperature is only applicable when creating a file. So I guess this would be destination temperature. That also means that the new CopyFile overload above doesn't need the temperature argument, since the dest file is already created.

Reply:
@pdillinger My latest commit attempting to fix this is causing test failures. Either my understanding of temperature or the test is incorrect. In any case, I'll revert my commit and defer this to someone more knowledgeable about the feature.